src/service/action-impl.js

```diff
@@ -924,14 +946,21 @@ export function parseActionMap(action, context, opt_defaultArgs) {
         let args = null;
 
         peek = toks.peek();
-        if (peek.type == TokenType.SEPARATOR && peek.value == '.') {
-          toks.next(); // Skip '.'
-          method = assertToken(
-              toks.next(), [TokenType.LITERAL, TokenType.ID]).value || method;
-
-          // Optionally, there may be arguments: "(key = value, key = value)".
-          peek = toks.peek();
-          if (peek.type == TokenType.SEPARATOR && peek.value == '(') {
+        if (peek.type == TokenType.SEPARATOR
+            && (peek.value == '.' || peek.value == '(')) {
+          if (peek.value == '.') {
+            toks.next(); // Skip '.'
+            method = assertToken(
+                toks.next(), [TokenType.LITERAL, TokenType.ID]).value || method;
+
+            // Optionally, there may be arguments: "(key = value, key = value)".
+            peek = toks.peek();
+            if (peek.type == TokenType.SEPARATOR && peek.value == '(') {
+              toks.next(); // Skip '('
+              args = tokenizeMethodArguments(toks, assertToken, assertAction);
+            }
+          } else {
+            // This is an action-macro invocation.
```

Comment:
I'm not sure we need to support arguments without a method e.g. `on="tap:myMacro(foo=1)"`. I wonder if the syntax may be confusing.

Instead of changing the syntax structure, we could just register a non-default action in amp-action-macro.js like "execute".

Reply:
So this is similar to how we invoke default actions on a component, in this case the targeted component is the amp-action macro but yes what is confusing is that now we are now passing in arguments to the default action without it being explicit in the markup. I think registering the 'execute' alias action is a good idea. Thanks